src/redux/message-service/saga.ts

```diff
@@ -61,10 +69,44 @@ function* receiveFromFirebaseSaga(address: string) {
   }
 }
 
+function* validateMessage(data, signature) {
+  const appState: ApplicationState = yield select(getApplicationState);
+  let opponentIndex = 0;
+  if (appState.gameState) {
+    opponentIndex = 1 - appState.gameState.playerIndex;
+  }
+  const requestId = hash(data + Date.now());
+  yield put(walletActions.validationRequest(requestId, data, signature, opponentIndex));
+  const actionFilter = [walletActions.VALIDATION_SUCCESS, walletActions.VALIDATION_FAILURE];
+  let action: walletActions.ValidationResponse = yield take(actionFilter);
+  while (action.requestId !== requestId) {
+    action = yield take(actionFilter);
+  }
+  if (action.type === walletActions.VALIDATION_SUCCESS) {
+    return true;
+  } else {
+    // TODO: Properly handle this.
+    throw new Error("Signature Validation error");
+  }
+}
+
+function* signMessage(data) {
+  const requestId = hash(`${data}${Date.now()}`);
```

Comment:
Is this different to the `hash(data + Date.now())` used on line 78?

Reply:
Nope, they should both be done the same way for consistency